### PR TITLE
Add files and instructions for using Mountpoint in Docker

### PR DESCRIPTION
**Pull request by mheshwg**
Mountpoint for Amazon S3 is dockerised with instruction to run the same.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

**Review by monthonk — CHANGES_REQUESTED** [review]
Thanks for trying it out on Docker and contributing to the project! I have a few comments on what we should consider.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#THIS CONTENT IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+#KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+#WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+#PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL AWS BE
+#LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN
+#AN ACTION OF CONTRACT, TORT, OR OTHERWISE, ARISING FROM, OUT
+#OF, OR IN CONNECTION WITH THE CONTENT OR THE USE OF THE
+#CONTENT
+
+FROM amazonlinux:latest
```
[the remark this answers is not in the document]
Comment (dannycjones) [reply Dockerfile:18]: If you do continue to use Amazon Linux, please also use a stable major version tag from the AWS ECR repository such as `public.ecr.aws/amazonlinux/amazonlinux:2023`. (more info on the ECR image: https://gallery.ecr.aws/amazonlinux/amazonlinux)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+#SUBJECT TO ANY APPLICABLE AGREEMENT BETWEEN YOU AND AWS,
+#THIS CONTENT IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+#KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+#WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+#PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL AWS BE
+#LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN
+#AN ACTION OF CONTRACT, TORT, OR OTHERWISE, ARISING FROM, OUT
+#OF, OR IN CONNECTION WITH THE CONTENT OR THE USE OF THE
+#CONTENT
```
[the remark this answers is not in the document]
Comment (mheshwg) [reply Dockerfile]: Removed the copyright text

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (mheshwg) [reply Dockerfile]: Included base image amazonlinux:2023 stable version.

File: Dockerfile
[the remark this answers is not in the document]
Comment (mheshwg) [reply Dockerfile:1]: Created a directory docker to hold Dockerfile. good one thank you.

File: README.md [hunk 3]
```diff
@@ -47,6 +47,17 @@ Then run the client, specifying the directory in which your S3 bucket should be
 
 The client will run in the background by default, and the `~/mnt` directory now gives access to the objects in your S3 bucket.
 
+### Mountpoint for Amazon S3 using Docker
+
+First, build the docker image using the below command.
+    
+    cd mountpoint-s3
+    docker build mount-s3 .
```
[the remark this answers is not in the document]
Comment (mheshwg) [reply README.md]: Fixed it. missing the `-t`

File: README.md [hunk 4]
```diff
@@ -47,6 +47,17 @@ Then run the client, specifying the directory in which your S3 bucket should be
...
+
+Once docker build is successful, run the below command to mount s3 and list objects 
+
+     docker run -it --cap-add SYS_ADMIN --device /dev/fuse --env AWS_ACCESS_KEY_ID=<<KEY ID>> --env AWS_SECRET_ACCESS_KEY=<<Secret>> --env AWS_DEFAULT_REGION=<<AWS Region>> mount_s3 /bin/bash -c "./mount-s3 <<Bucket name>> /mnt; ls -l /mnt"
```
[the remark this answers is not in the document]
Comment (mheshwg) [reply README.md]: Good one. fixed it.

[the remark this answers is not in the document]
Comment (mheshwg) [reply README.md]: Fixed it , suggested env variables and included the variable name as run time. Good one!.

**Comment by mheshwg** [comment]
Incorporated the review comments provided. please let me know if any more comments.

**Review by dannycjones — CHANGES_REQUESTED** [review]
Just a small note to enable use of short-term credentials. Looking to get this merged tomorrow if ready.

File: README.md [hunk 5]
```diff
@@ -47,6 +47,24 @@ Then run the client, specifying the directory in which your S3 bucket should be
...
+Once docker build is successful, Set AWS credentials as environmental variables as specified in https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
+Run the below command to mount s3 and list objects. Please fill the <<Bucket name>> before executing the below command.
+ 
+     docker run -it \
+        --cap-add SYS_ADMIN \
+        --device /dev/fuse \
+        --env AWS_ACCESS_KEY_ID \
+        --env AWS_SECRET_ACCESS_KEY \
+        --env AWS_DEFAULT_REGION \
+        mount-s3 /bin/bash -c "./mount-s3 <<Bucket name>> /mnt; ls -l /mnt"
```
[the remark this answers is not in the document]
Comment (mheshwg) [reply README.md]: Included the recommended changes. Kindly merge for others to access.

**Comment by dannycjones** [comment]
DCO check in CI will specify which commits need the sign-off trailer added. You can use an interactive rebase to pause to edit each commit listed, you run `git commit --amend --signoff --no-edit` on each commit it pauses at and, when complete, force push over your fork's branch.

**Review by dannycjones — APPROVED** [review]
Approved, but needs DCO fix on commits.

**Comment by mheshwg** [comment]
> Is the source branch correct? Looks like there are some code changes included.

are you referring to *.rs files under mountpoint-s3/src?. I am not sure how these files are in.. these files are not part of my changes.. how do we remove those in this pull request?.. Is the change align with the main branch of yours?

**Comment by dannycjones** [comment]
I'd propose to move your changes to a new branch on your fork and drop any commits that shouldn't be there, like the last two currently (6e0057a, a417443).

**Comment by monthonk** [comment]
Replaced by https://github.com/awslabs/mountpoint-s3/pull/185

**Pull request closed without merging**